### DISCUSSION TITLE
Upgrading to Yeti v2

### DISCUSSION THREAD

**Discussion by udgover** [yeti-platform/pyeti#24] [General]
Let's discuss what should be implemented in pyeti client to support Yeti v2 and how it should be implemented.

I've already started to implement some functions to use from a Jupyter notebook and I'd like to discuss how to merge this work in pyeti.

**Comment by tomchop** [comment]
The way I see it is:

* Have a "base module" with one python function for each API endpoint
* Have a "extended" module for functions that combine any of the previous ones (e.g. "observables for campaign" or something)
* Design the base module in a way where the connector (i.e. the function that does the actual HTTP request) can be easily swapped (so we can add headers / sessionize everything, etc.)
